packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/LegacyManifest.java

```diff
@@ -158,39 +158,43 @@ public UpdateEntity getUpdateEntity() {
 
   private Uri getAssetsUrlBase() {
     if (mAssetsUrlBase == null) {
-      String hostname = mManifestUrl.getHost();
-      if (hostname == null) {
-        mAssetsUrlBase = Uri.parse(EXPO_ASSETS_URL_BASE);
-      } else {
-        for (String expoDomain : EXPO_DOMAINS) {
-          if (hostname.contains(expoDomain)) {
-            mAssetsUrlBase = Uri.parse(EXPO_ASSETS_URL_BASE);
-            break;
-          }
+      mAssetsUrlBase = getAssetsUrlBase(mManifestUrl, getRawManifestJson());
+    }
+    return mAssetsUrlBase;
+  }
+
+  /* package */ static Uri getAssetsUrlBase(Uri manifestUrl, JSONObject manifestJson) {
+    String hostname = manifestUrl.getHost();
+    if (hostname == null) {
+      return Uri.parse(EXPO_ASSETS_URL_BASE);
+    } else {
+      for (String expoDomain : EXPO_DOMAINS) {
+        if (hostname.contains(expoDomain)) {
+          return Uri.parse(EXPO_ASSETS_URL_BASE);
         }
+      }
 
-        if (mAssetsUrlBase == null) {
-          // assetUrlOverride may be an absolute or relative URL
-          // if relative, we should resolve with respect to the manifest URL
-          String assetsPathOrUrl = getRawManifestJson().optString("assetUrlOverride", "assets");
-          Uri maybeAssetsUrl = Uri.parse(assetsPathOrUrl);
-          if (maybeAssetsUrl != null && maybeAssetsUrl.isAbsolute()) {
-            mAssetsUrlBase = maybeAssetsUrl;
-          } else {
-            // use manifest URL as the base
-            Uri.Builder assetsBaseUrlBuilder = mManifestUrl.buildUpon();
-            List<String> segments = mManifestUrl.getPathSegments();
-            assetsBaseUrlBuilder.path("");
-            for (int i = 0; i < segments.size() - 1; i++) {
-              assetsBaseUrlBuilder.appendPath(segments.get(i));
-            }
-            assetsBaseUrlBuilder.appendPath(assetsPathOrUrl);
-            mAssetsUrlBase = assetsBaseUrlBuilder.build();
-          }
+      // assetUrlOverride may be an absolute or relative URL
+      // if relative, we should resolve with respect to the manifest URL
+      String assetsPathOrUrl = manifestJson.optString("assetUrlOverride", "assets");
+      Uri maybeAssetsUrl = Uri.parse(assetsPathOrUrl);
+      if (maybeAssetsUrl != null && maybeAssetsUrl.isAbsolute()) {
+        return maybeAssetsUrl;
+      } else {
+        if (assetsPathOrUrl.startsWith("./")) {
+          assetsPathOrUrl = assetsPathOrUrl.substring(2);
         }
```

Comment:
java.net.URI has `normalize` which we might want to use here and then convert to an Android Uri. This will take care of `../` as well.